### PR TITLE
Minimal self-contained docker image

### PR DESCRIPTION
**Pull request by rvl**
### Overview

Adds a Dockerfile which builds a super-minimal Docker image for cardano-wallet-shelley, where everything except the nixpkgs base is built from source.

### Comments

```
rodney@blue:~ % docker run --rm 8e585b18d3e9                                                      
The CLI is a proxy to the wallet server, which is required for most commands.
Commands are turned into corresponding API calls, and submitted to an
up-and-running server. Some commands do not require an active server and can be
run offline (e.g. 'mnemonic generate').

Usage: cardano-wallet-shelley COMMAND
  Cardano Wallet Command-Line Interface (CLI)

Available options:
  -h,--help                Show this help text

Available commands:
  serve                    Serve API that listens for commands/actions.
  mnemonic                 Manage mnemonic phrases.
  key                      Derive and manipulate keys.
  wallet                   Manage wallets.
  address                  Manage addresses.
  transaction              Manage transactions.
  network                  Manage network.
  stake-pool               Manage stake pools.
  version                  Show the program's version.
rodney@blue:~ % docker images | grep 8e585b18d3e9                                                 
<none>              <none>              8e585b18d3e9        2 hours ago         40.6MB
rodney@blue:~ %                    
```

### CODE REVIEW

File: nix/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+############################################################################
+# Create final container
+
+FROM scratch
+CMD mkdir -p bin
+# It is ok to COPY files from a build container (when using multi-stage builds)
+COPY --from=builder out/cardano-node bin/cardano-node
+COPY --from=builder out/cardano-wallet-shelley bin/cardano-wallet-shelley
+ENTRYPOINT ["cardano-wallet-shelley"]
+CMD []
```
Comment (KtorZ) [inline nix/Dockerfile:83]: Sweet.

File: nix/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+RUN nix-build https://github.com/input-output-hk/cardano-wallet/archive/master.tar.gz -A cardano-wallet-shelley --no-out-link
+
+
+############################################################################
+# Check out code
+
+# You can choose a branch, tag, or git revision of the cardano-wallet repo.
+# fixme: temporary PR branch
+ARG CODE_VERSION=rvl/minimal-self-contained-docker-image
+# ARG CODE_VERSION=master
```
Comment (KtorZ) [inline nix/Dockerfile:55]: `master` before merging huh :sweat_smile: ?

File: nix/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# attribute and copy that across.
+
+
+############################################################################
+# Create final container
+
+FROM scratch
+CMD mkdir -p bin
+# It is ok to COPY files from a build container (when using multi-stage builds)
+COPY --from=builder out/cardano-node bin/cardano-node
```
Comment (KtorZ) [inline nix/Dockerfile:80]: Why `cardano-node` here? There's no `launch` from `cardano-wallet-shelley`.

Comment (rvl) [reply nix/Dockerfile:80]: I think we should consider adding cardano-node launching to cardano-wallet-shelley.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**